### DISCUSSION TITLE
Let's Get a Live Mode Rolling in This Framework!

### DISCUSSION THREAD

**Discussion by adnumaro** [nuxt/nuxt#23535] [Ideas]
Hey Everyone,

Hope you’re doing good. Been tinkering with Nuxt for a while now, and it’s pretty slick for devs, ain’t it? But something’s been bugging me, and I wanna toss it out here to see what you all think.

Ever thought about throwing a "Live" mode into the mix? Kinda like what [LiveView](https://hexdocs.pm/phoenix_live_view/Phoenix.LiveView.html) does in the Phoenix framework (Elixir), or what's up with [LiveView.js](https://github.com/floodfx/liveviewjs)?

Integrating a feature like this could seamlessly unlock opportunities for developing real-time applications, while maintaining the user-friendly developer experience characteristic of Nuxt

I reckon the community and I would get a kick out of batting this idea around, discussing potential implementation, and hashing out any challenges that might pop up.

Thanks a bunch for mulling over this idea. Chomping at the bit to hear your thoughts and feedback.

**Comment by Hebilicious** [comment]
Hi, what's the difference with the current HMR ?

**Comment by TheAlexLichter** [comment]
I think this is what server components will allow us to do eventually ☺️

**Comment by adnumaro** [comment]
The difference is that LiveView works with websockets and not with http calls. Additionally, LiveView has features like Presence, which allows knowing which users are currently connected to a specific socket. The Phoenix library's cardinal component is channels. It allows millions of connected clients to communicate in soft real-time.

Currently, this can be done in Nuxt, since you can add websockets to the server. However, if you want complete real-time functionality, it's quite more complicated. There are several libraries that make it possible, such as LiveBlocks.js or TiptapCollaboration with the Hocuspocus server or even Supabase.

The problem itself lies not only in the dependency on an external server, which you cannot control but also in several related challenges. One significant concern is data security, ensuring that sensitive and crucial information is thoroughly protected and secure during transmission and storage on these external servers. Determining which server acts as the 'source of trust' can also be challenging and vital for maintaining data integrity across platforms. Furthermore, disparities between different database systems and structures may present integration issues or inconsistencies in data management.

In real-time operations, data on an owned server can quickly become outdated, which is crucial if your application relies on up-to-the-minute information. There may be instances where specific features on your server require that data be updated at that exact moment, necessitating not just accurate, but also timely data synchronization. Speed and performance are also paramount. Slower data updates or transfers can hamper user experience, especially in applications where real-time updates are crucial for functionality. Managing and ensuring synchronization through Webhooks while maintaining speed, performance, and security can become quite the challenge, especially for larger applications, turning it into a significant problem.
